sql-common/client.c

```diff
@@ -1897,15 +1897,27 @@ mysql_get_ssl_cipher(MYSQL *mysql __attribute__((unused)))
 
 #if defined(HAVE_OPENSSL) && !defined(EMBEDDED_LIBRARY)
 
+#include <openssl/x509v3.h>
+
+#if !defined(HAVE_YASSL) && !defined(LIBRESSL_VERSION_NUMBER)
+  #if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10002000L
+    #define HAVE_X509_check_host 1
+  #endif
+#endif
+
 static int ssl_verify_server_cert(Vio *vio, const char* server_hostname, const char **errptr)
```

Comment:
General remark on this function - it is huge. Would you consider splitting it into smaller functions - it would be much easier to read. There is quite a distinction between SAN validation and CN validation.